go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -2016,12 +2019,20 @@ func setNodeMasqueradeIPOnExtBridge(extBridgeName string) error {
 	if config.IPv6Mode {
 		_, masqIPNet, _ := net.ParseCIDR(config.Gateway.V6MasqueradeSubnet)
 		masqIPNet.IP = config.Gateway.MasqueradeIPs.V6HostMasqueradeIP
-		bridgeCIDRs = append(bridgeCIDRs, cidrAndFlags{ipNet: masqIPNet, flags: unix.IFA_F_NODAD})
+		// Deprecate the IPv6 host masquerade IP address to ensure its not used in source address selection except
+		// if a route explicitly sets its src IP as this masquerade IP. See RFC 3484 for more details for linux src address selection.
+		// Currently, we set a route with destination as the service CIDR with source IP as the host masquerade IP.
+		// Also, ideally we would only set the preferredLifetime to 0, but because this is the default value of this type, the netlink lib
+		// will only propagate preferred lifetime to netlink if either preferred lifetime or valid lifetime is set greater than 0.
+		// Set valid lifetime to max will achieve our goal of setting preferred lifetime 0.
+		bridgeCIDRs = append(bridgeCIDRs, cidrAndFlags{ipNet: masqIPNet, flags: unix.IFA_F_NODAD, preferredLifetime: 0,
```

Comment:
so the end result of this is, we end up actually setting the preferred lifetime to literally 0, so the address becomes deprecated right away, while the valid lifetime is max int, which is equivalent to "forever". Right @martinkennelly ? I'll checkout the code and launch kind too just to confirm.

Reply:
Looks to be the case: 
```
5: breth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 state UNKNOWN qlen 1000
    inet6 fd69::2/125 scope global nodad deprecated 
       valid_lft forever preferred_lft 0sec

```

Reply:
>so the end result of this is, we end up actually setting the preferred lifetime to literally 0, so the address becomes deprecated right away, while the valid lifetime is max int, which is equivalent to "forever". Right @martinkennelly ? 

yes